app/src/pages/trace/TracePage.tsx

```diff
@@ -1057,64 +1057,107 @@ function EmbeddingSpanInfo(props: {
 }
 
 function ToolSpanInfo(props: { span: Span; spanAttributes: AttributeObject }) {
-  const { spanAttributes } = props;
+  const { span, spanAttributes } = props;
+  const { input, output } = span;
+  const inputIsText = input?.mimeType === "text";
+  const outputIsText = output?.mimeType === "text";
   const toolAttributes = useMemo<AttributeTool>(
     () => spanAttributes[SemanticAttributePrefixes.tool] || {},
     [spanAttributes]
   );
   const hasToolAttributes = Object.keys(toolAttributes).length > 0;
-  if (!hasToolAttributes) {
-    return null;
-  }
   const toolName = toolAttributes[ToolAttributePostfixes.name];
   const toolDescription = toolAttributes[ToolAttributePostfixes.description];
   const toolParameters = toolAttributes[ToolAttributePostfixes.parameters];
+  if (
+    (!input || input.value == null) &&
+    (!output || output.value == null) &&
+    !hasToolAttributes
+  ) {
+    return null;
+  }
   return (
     <Flex direction="column" gap="size-200">
-      <Card
-        title={"Tool" + (typeof toolName === "string" ? `: ${toolName}` : "")}
-        {...defaultCardProps}
-      >
-        <Flex direction="column">
-          {toolDescription != null ? (
-            <View
-              paddingStart="size-200"
-              paddingEnd="size-200"
-              paddingTop="size-100"
-              paddingBottom="size-100"
-              borderBottomColor="dark"
-              borderBottomWidth="thin"
-              backgroundColor="light"
-            >
-              <Flex direction="column" alignItems="start" gap="size-50">
-                <Text color="text-700" fontStyle="italic">
-                  Description
-                </Text>
-                <Text>{toolDescription as string}</Text>
+      {input && input.value != null ? (
```

Comment:
```suggestion
      {hasInput? (
```
since this boolean operator is repeated I would just store it above and name it as so to make it a bit more readable